cmd/zq/ztests/single-arg-error.yaml

```diff
@@ -0,0 +1,11 @@
+script: |
+  ! zq 'file sample.zson | count('
+
+outputs:
+  - name: stderr
+    data: |
+      zq: could not invoke zq with a single argument because:
+       - a file could not be found with the name "file sample.zson | c..."
+       - the argument could not be compiled as a valid Zed query due to parse error (column 25):
+         file sample.zson | count(
+                              === ^ ===
```

Comment:
This is pretty wordy. How about something more concise?
```
zq: not a file or valid query: "from ( file a ) | co..."
  error parsing Zed at column 25:
  from ( file a ) | count(
                      === ^ ===
```

As a bonus, that would make for a much simpler `singleArgError`.
```go
func singleArgError(src string, err error) error {
	if len(src) > 20 {
		src = src[:20] + "..."
	}
	msg := fmt.Sprintf("not a file or valid query: %q", src)
	if errors.As(err, new(*parser.Error)) {
		msg += "\n" + err.Error()
	}
	msg = strings.ReplaceAll(msg, "\n", "\n   ")
	return errors.New(msg)
}
```

Reply:
(To be clear, I'm OK with the existing message if you guys prefer it.)

Reply:
Thanks @nwt. Indeed, I was in the middle of writing a reply when that last comment of yours came in, so I'll paste that below for completeness, but basically, yeah, I happen to prefer what's already there. 😄

I know I can tend toward wordiness in my own work and so I try to keep that in check, but I think this is one place where verbosity is maybe justified. For instance, in the #5046 case the user is almost certainly thinking entirely in terms of their argument being a Zed query so I'm concerned something concise like "not a file" might still be confusing. "Not a file? What? The file refs are in my program. I don't get it." This particular error is triggered rarely enough and has confused users enough in is previous form in enough that I feel pretty strongly that the benefits of super clear communication outweigh any assumed benefits from being concise.

Come to think of it, the user that last hit #5046 is responsive on Slack today, so I'll see if they're game to weigh in to the discussion, though if we already have consensus and want to get on with it and merge I'm fine with that too.

Reply:
@mattnibs: Any preference here?

Reply:
fwiw, as someone who doesn't use zed every day, but reaches for it probably once a week when manipulating data I'd prefer the more verbose message. It's hard to understand exactly what might be wrong, especially given the unfamiliarity of the zq syntax compared to other tools.